storage/mega/storage_mega.go

```diff
@@ -0,0 +1,277 @@
+			currentRoot = nodesInCurrentRoot[0]
+		}
+	}
+	return nil
+}
+
+// Stat returns the size of a file
+func (backend *StorageMega) Stat(path string) (uint64, error) {
+	node, err := backend.GetLastNodeInPath(path)
+	if err != nil {
+		return 0, err
+	}
+	return uint64(node.GetSize()), nil
+}
+
+// ReadFile reads a file from mega
+func (backend *StorageMega) ReadFile(path string) ([]byte, error) {
+
+	nodeToRead, err := backend.GetLastNodeInPath(path)
+	if err != nil {
+		return nil, err
+	}
+
+	download, err := backend.mega.NewDownload(&nodeToRead)
+	if err != nil {
+		return nil, err
+	}
+	var bytes []byte
+
+	for i := 0; i < download.Chunks(); i++ {
+		chunkBytes, err := download.DownloadChunk(i)
+		if err != nil {
+			return nil, err
+		}
+		bytes = append(bytes, chunkBytes...)
+	}
+
+	download.Finish()
+	return bytes, nil
+}
+
+// WriteFile writes file on mega
+func (backend *StorageMega) WriteFile(path string, data []byte) (size uint64, err error) {
+	var progress chan int
+	dir, file := filepath.Split(path)
+
+	_, err = backend.GetLastNodeInPath(path)
+	if err == nil {
+		// sadly, if the file exists it needs to be deleted before re-uploading, otherwise there will be a copy
+		err = backend.DeleteFile(path)
+		if err != nil {
+			return 0, err
+		}
+	}
+	nodeToWriteIn, err := backend.GetLastNodeInPath(dir)
+	if err != nil {
+		return 0, err
+	}
+	upload, err := backend.mega.NewUpload(&nodeToWriteIn, file, int64(len(data)))
+	if err != nil {
+		return 0, err
+	}
+
+	// following code is based on https://sourcegraph.com/github.com/t3rm1n4l/go-mega@79a813bb328ddee4889f53a97fbdcd71e5ae1a09/-/blob/mega.go#L1532:16
+	workch := make(chan int)
+	errch := make(chan error, 1)
+	wg := sync.WaitGroup{}
+	workers := runtime.NumCPU()
+
+	// fire chunk upload workers
+	for w := 0; w < workers; w++ {
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+
+			for id := range workch {
+				chk_start, chk_size, err := upload.ChunkLocation(id)
+				if err != nil {
+					errch <- err
+					return
+				}
+				err = upload.UploadChunk(id, data[chk_start:chk_start+int64(chk_size)])
+				if err != nil {
+					errch <- err
+					return
+				}
+
+				if progress != nil {
+					progress <- chk_size
+				}
+			}
+		}()
+	}
+
+	// place chunk download jobs to chan
+	err = nil
+	for id := 0; id < upload.Chunks() && err == nil; {
+		select {
+		case workch <- id:
+			id++
+		case err = <-errch:
+		}
+	}
+
+	close(workch)
+	wg.Wait()
+
+	if err != nil {
+		return 0, err
+	}
+
+	upload.Finish()
+	return uint64(len(data)), err
+}
+
+// DeleteFile deletes a file from mega
+func (backend *StorageMega) DeleteFile(path string) error {
+	fileToDelete, err := backend.GetLastNodeInPath(path)
+	if err != nil {
+		return err
+	}
+	err = backend.mega.Delete(&fileToDelete, true)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+// GetLastNodeInPath() returns the last node in a path on mega. It may be a file or a directory node.
```

Comment:
I think we wanna call this method `getNodeFromPath` and keep it private. It should only be used internally, after all.